### ISSUE TITLE
MuseScore crash when pushing the "eye" to hide a stave

### ISSUE THREAD

**Issue by pacebes** [musescore/MuseScore#24657]
### Issue type

Crash or freeze

### Description with steps to reproduce

1. Open the attached file MuseScore file (MuseScore crashes when hiding a stave)
2. Click in the "eye" to hide a Stave (any of them)
![MuseScore breaks when clocking in the eye to hide a Stave](https://github.com/user-attachments/assets/e5e13592-f2a9-4ced-b4ef-d6b9b0fee0f4)
3. MuseScore crashes as it can be seen on the Video

### Supporting files, videos and screenshots

[MuseScore crashes when hiding a stave.zip](https://github.com/user-attachments/files/16976306/MuseScore.crashes.when.hiding.a.stave.zip)

https://github.com/user-attachments/assets/944a5a3e-0edd-45c4-8d44-476b66286165

### What is the latest version of MuseScore Studio where this issue is present?

OS: Windows 11 Version 2009 or later, Arch.: x86_64, MuseScore Studio version (64-bit): 4.4.1-242490810, revision: github-musescore-musescore-0b3dd00

### Regression

I was unable to check

### Operating system

Windows 11

### Additional context

- It also happens with nightly build OS: Windows 11 Version 2009 or later, Arch.: x86_64, MuseScore Studio version (64-bit): 4.4.2-242530503, revision: github-musescore-musescore-17a8f8e
- The Score was imported from a XML exported by Encore (I can provide you with the xml if necessary)
- Everything else I have tried with MuseScore works fine with this Score
- I have searched for similar bugs, but I couldn't find anything similar ( https://github.com/musescore/MuseScore/issues?q=is%3Aissue+is%3Aopen+crash+hide+stave )

### Checklist

- [X] This report follows the [guidelines](https://github.com/musescore/MuseScore/wiki/Reporting-bugs-and-issues) for reporting bugs and issues
- [X] I have verified that this issue has not been logged before, by searching the [issue tracker](https://github.com/musescore/MuseScore/issues) for similar issues
- [X] I have attached all requested files and information to this report
- [X] I have attempted to identify the root problem as concisely as possible, and have used minimal reproducible examples where possible

**Comment by MarcSabatella** [comment]
FWIW, I can't reproduce on Linux with either 4.4.1 or the most recent 4.4.2 nightly, but I can on Windows.

**Comment by bkunda** [comment]
I can only reproduce using this score (on macOS), so we will probably need the diagnostic files to investigate. 

Please upload them here (get them from Diagnostic > Save diagnostic files).

**Comment by pacebes** [comment]
HI

I attach two files:

1. Open the score "MuseScore crashes when hiding a stave"
2. Generate the diagnostic file "Crashing when hiding a stave MuseScore diagnostic file BEFORE generating crash.zip"


[Crashing when hiding a stave MuseScore diagnostic file BEFORE generating crash.zip](https://github.com/user-attachments/files/16997139/Crashing.when.hiding.a.stave.MuseScore.diagnostic.file.BEFORE.generating.crash.zip)

3. Click on the "eye" and generate a crash
4. Open the file again and generate the diagnostic file "Crashing when hiding a stave MuseScore diagnostic file AFTER generating crash.zip"

[Crashing when hiding a stave MuseScore diagnostic file AFTER generating crash.zip](https://github.com/user-attachments/files/16997154/Crashing.when.hiding.a.stave.MuseScore.diagnostic.file.AFTER.generating.crash.zip)

I hope it helps!.

Anyway, If this is something really really rare it doesn't need to be fixed. I can create this Score from the scratch in less than 15 minutes.

Thanks.

**Comment by cbjeukendrup** [comment]
I investigated the crash; it has to do with the voltas. At https://github.com/musescore/MuseScore/blob/d9a7f51e22f4780ff318e9677f8a4a9e6a28de2f/src/engraving/dom/volta.cpp#L363
`staffIdxOrNextVisible()` is `muse::nidx`. The volta in question has `track = 4` at that point, i.e. second staff, but I don't know if that is because it really belongs to the second staff, or because it has been moved there. 

Of course, some check should be added there, to verify that `staffIdxOrNextVisible()` is not `muse::nidx` before passing it to `segment->staffShape`. But for really _handling_ that case, it is already too late when `Volta::linePos` is called, because if I am not mistaken that should mean that the volta is not going to be visible / laid out at all. 
It seems that this can be achieved by adding 
```diff
         if (!sp->systemFlag() && sp->staff() && !sp->staff()->show()) {
             continue;
         }
+        if (sp->systemFlag() && sp->staffIdxOrNextVisible() == muse::nidx) {
+            continue;
+        }
``` 
in `SystemLayout::layoutSystemElements`, but I'm not sure if that's the best thing to do.
